Makefile

```diff
@@ -35,25 +35,38 @@ ifeq ($(shell cd $(KERNELPATH) && pwd),)
 $(warning $(KERNELPATH) is missing, please set KERNELPATH)
 endif
 
+OBJCOPY ?= objcopy
+OBJDUMP ?= objdump
+
 export KERNELPATH
 RM ?= rm -f
 
 REVISION= $(shell	if [ -d "$(PWD)/.git" ]; then \
 				echo $$(git --git-dir="$(PWD)/.git" describe --always --dirty --match "v*" |sed 's/^v//' 2> /dev/null || echo "[unknown]"); \
 			fi)
 
-CONFIG_BATMAN_ADV=m
-batman-adv-y += compat.o
+CONFIG_BATMAN_ADV_LEGACY=m
+batman-adv-legacy-y += compat.o
 ifneq ($(REVISION),)
 ccflags-y += -DBATADV_SOURCE_VERSION=\"$(REVISION)\"
 endif
 include $(PWD)/Makefile.kbuild
 
-all: config
+all: build updatesyms
+
+build: config
 	$(MAKE) -C $(KERNELPATH) M=$(PWD) PWD=$(PWD) modules
 
+updatesyms:
+	@if ! $(OBJDUMP) -j .init.text -t batman-adv-legacy.ko| grep -q "batadv_legacy_init"; then \
+		$(OBJDUMP) -t batman-adv-legacy.ko | grep batadv_ | grep -v "batadv_legacy" | \
+			sed "s/.* \([^ ]*\)batadv_\([^ ]*\)$$/\1batadv_\2 \1batadv_legacy_\2/" \
+				>> $(PWD)/updated-syms.txt; \
+		$(OBJCOPY) --redefine-syms=$(PWD)/updated-syms.txt $(PWD)/batman-adv-legacy.ko; \
```

Comment:
Using the same filename for input and output of the objcopy is making this more fragile than necessary. Just use *batman-adv.ko* as input, *batman-adv-legacy.ko* as output, and drop the whole if conditional.

We can use proper file targets then, which will also improve error handling:
- rename build to batman-adv.ko
- rename updatesyms to batman-adv-legacy.ko
- create a proper target to updated-syms.txt
- properly define dependencies between the different targets